### PR TITLE
orchard: init at 0.11.0

### PR DESCRIPTION
**Pull request by roblabla**
## Description of changes

This creates a package for `orchard`, an orchestrator for running Virtual Machines on a cluster of Apple Silicon devices.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -42238,4 +42238,6 @@ with pkgs;
   ssl-proxy = callPackage ../tools/networking/ssl-proxy { };
 
   code-maat = callPackage ../development/tools/code-maat {};
+
+  orchard = callPackage ../applications/virtualization/orchard { };
```
Comment (zeuner) [inline pkgs/top-level/all-packages.nix:42242]: How about switching to the new `by-name` mechanism (https://github.com/NixOS/nixpkgs/pull/237439)?

File: pkgs/applications/virtualization/orchard/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    owner = "cirruslabs";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-DRaolnvHpijnC1LuQlJCG3xj7cU6YJsbCxYrB22E82U=";
+  };
+  vendorHash = "sha256-BrzS+QtpGUHcYNNmSI6FlBtcYwNFri7R6nlVvFihdb4=";
+
+  nativeCheckInputs = [ tart ];
+  # This has a ton of failures I cannot explain.
+  doCheck = false;
```
Comment (zeuner) [inline pkgs/applications/virtualization/orchard/default.nix:15]: Is it possible to disable the failing tests individually, so it's easier and more obvious to address them later on?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (roblabla) [reply pkgs/top-level/all-packages.nix:42242]: Oh, TIL. I'll do that.

**Comment by roblabla** [comment]
This got packaged in another PR.

**Pull request closed without merging**